x-pack/libbeat/common/cloudfoundry/dopplerconsumer.go

```diff
@@ -80,23 +81,49 @@ func (c *DopplerConsumer) Run() {
 }
 
 func (c *DopplerConsumer) logsFirehose() {
-	c.firehose(c.callbacks.Log, consumer.LogMessages)
+	filter := FirehoseFilterAllLogs
+	if c.callbacks.Filter != "" {
+		filter = c.callbacks.Filter
+	}
+	filterFn, envelopeFilter := selectFilter(filter)
+	c.firehose(c.callbacks.Log, filterFn, envelopeFilter)
 }
 
 func (c *DopplerConsumer) metricsFirehose() {
-	c.firehose(c.callbacks.Metric, consumer.Metrics)
+	filter := FirehoseFilterMetrics
+	if c.callbacks.Filter != "" {
+		filter = c.callbacks.Filter
+	}
+	filterFn, envelopeFilter := selectFilter(filter)
+	c.firehose(c.callbacks.Metric, filterFn, envelopeFilter)
+}
+
+func selectFilter(firehoseFilter string) (func(*events.Envelope) bool, consumer.EnvelopeFilter) {
+	switch firehoseFilter {
+	case FirehoseFilterAll:
+		return filterNoFilter, -1
+	case FirehoseFilterAllLogs:
+		// Requests all events, and selects log-like events.
+		return filterLogs, -1
+	case FirehoseFilterLogs:
+		// Uses filter-type=logs in requests to the firehose.
+		return filterLogs, consumer.LogMessages
+	case FirehoseFilterMetrics:
+		// Uses filter-type=metrics in requests to the firehose.
+		return filterNoFilter, consumer.Metrics
+	default:
+		// TODO: Handle unknown filters.
+		return filterNoFilter, -1
```

Comment:
We should decide if we want some kind of error handling here, or if we just consider `all` the default in this case.